src/toil/test/src/retainTempDirTest.py

```diff
@@ -0,0 +1,37 @@
+# Copyright (C) 2015 UCSC Computational Genomics Lab
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from __future__ import absolute_import, print_function
+import os
+from toil.job import Job
+from toil.test import ToilTest
+
+class TempFileRetentionTest(ToilTest):
+    """
+    Tests testing the Job.FileStore class
+    """
+    def testRetainTempDir(self):
+        testDir = self._createTempDir()
+        options = Job.Runner.getDefaultOptions(self._getTestJobStorePath())
+        options.workDir = testDir
+        options.clean = "always"
+        options.cleanWorkDir = "never"
+        A = Job.wrapJobFn(tempFileTestJob)
+        Job.Runner.startToil(A, options)
+        retainedTempData = os.listdir(testDir)
+        assert retainedTempData != []
+
```

Comment:
Now that you implemented all four cases, shouldn't there be a test for each one of them?